anime-list-master.xml

```diff
@@ -21272,7 +21272,7 @@
   <anime anidbid="7729" tvdbid="244061" defaulttvdbseason="1" episodeoffset="" tmdbid="" imdbid="">
     <name>Steins;Gate</name>
     <mapping-list>
-      <mapping anidbseason="0" tvdbseason="0">;2-7;</mapping>
+      <mapping anidbseason="0" tvdbseason="0">;1-1;2-7;</mapping>
```

Comment:
1-1, 2-2, 3-3, etc.. Mapping should be implied \ automatic 🤔

Reply:
From what you have written previously, it sounds like season 0 still needs to be explicitly mapped though since several anidbids share this one tvdbid. Or does the '\<mapping anidbseason="0" tvdbseason="0">' section already implicitly map all specials one-to-one? If the latter, then yes 1-1 should be removed but 3-0 should be added.

Reply:
IIRC it _should_ map the whole season with a mapping-list present. But tbh I've been bit enough times by how HAMA says things should work and how they actually work

Reply:
I mean yeah, as I mentioned in the other thread HAMA doesn't even skip automapping when it's supposed to for series such as these. If that was fixed I could maybe figure out whether S1 or S3 gets mapped in this scenario, but right now it would just map both regardless. Which I guess does point to unmapping 3-0 being the solution, but that might not be the solution if HAMA gets fixed, and even then I think force-unmapping is broken in HAMA too. _sigh_

Reply:
Ah now I remember, take a look at your previous explanation for removing ;2-0;3-0; unmappings for anidbid="8556"
https://github.com/Anime-Lists/anime-lists/pull/179#discussion_r987294571
> Because of AniDB 8625, just `;1-2;` would do

Either:
Your previous comment a few months ago is correct, and unmapping is not necessary because the parser only maps eps explicitly listed. In this case an explicit 1-1 mapping for Steins;Gate is still needed.
or
Your above comment is correct, and the parser implicitly maps all eps in the season due to the existence of the mapping list. In which case unmapping needs to occur (3-0 for Steins;Gate, 2-0 for Another, 13-0 for SAO, etc.).
or
Something changed in the last 3 months.

Either way, I would agree ;3-0; unmapping was unnecessary in Another's case because there is no TVDB S00E03 to map to anyway.
I went the first route because of that existing precedent (admittedly I never looked through any other closed PRs, just that open one). I don't care either way, it just seems to me you can't have it both ways. Or maybe you can, but that would be by (using Steins;Gate for example) adding both ;1-1; explicit mapping and ;3-0; unmapping, not leaving them both off. So let me know which way you want it, but I do think SOMETHING should be updated for this entry.

Reply:
Unmapping _shouldn't_ be necessary, but turns out HAMA doesn't always behave as it says on the tin, so nowadays I'm not against a "better safe than sorry" approach xD

> But tbh I've been bit enough times by how HAMA says things should work and how they actually work

Reply:
Unmapping _shouldn't_ be necessary only if the parser doesn't automatically link 3-3, right? And if 3-3 isn't automatically linked, then similarly 1-1 is not automatically linked, and therefore should be listed in order to be mapped, as I have done here. So I believe this edit is valid, or is that incorrect?
And yeah that is how HAMA _should_ work, but as mentioned that isn't how it _actually_ works currently, but if this list has been built all this time under that assumption then it's probably still easier to try to get HAMA working as it should (and edit the list assuming it does) than have to deal with a bunch of erroneous entries...

Reply:
It is a fair assumption, I was just wondering if it wasn't working as expected currently to need it

Reply:
Sure fair assumption, we could unmap it to force it to work _now_ but that would just be a band-aid on a few shows when likely hundreds are affected, so... I'm not going to start unmapping now and instead focus effort on trying to get HAMA to work as expected lol.